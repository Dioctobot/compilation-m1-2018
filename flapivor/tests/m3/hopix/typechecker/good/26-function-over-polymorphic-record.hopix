type P<a, b> = { x : a; y : b }

def get_x : forall a, b. P<a, b> -> a (p) = p.x

def get_y : forall a, b. P<a, b> -> b (p) = p.y

val p : P<Int, String> = { x = 37; y = "Rory Williams" }<Int, String>

val x : Int = get_x<Int, String>(p)

val y : String = get_y<Int, String>(p)