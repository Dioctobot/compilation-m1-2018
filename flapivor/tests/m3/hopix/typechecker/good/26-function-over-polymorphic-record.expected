get_x : forall a, b. P<a, b> -> a
get_y : forall a, b. P<a, b> -> b
p : P<Int, String>
x : Int
y : String
